### PR TITLE
TMDM-12742: Externalize UserQuery in module

### PR DESCRIPTION
**Pull request by YohannTalend**
https://jira.talendforge.org/browse/TMDM-12742

Isolate User query construction into a module to allow it to be reused from an external perspective

The goal is to put the changes on the repository MDM Server all the changes developed to externalize the code. And then, to verify the compilation and the jenkins status.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [] Yes
- [x ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications:

This PR must be break something because it only contains a refactoring on the code to externalize the code of query for proxy side.

**Other information**:

### CODE REVIEW

File: org.talend.mdm.query/src/com/amalto/core/storage/record/CommonStorage.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.amalto.core.storage.record;
+
+public interface CommonStorage {
```
Comment (pbabin) [inline org.talend.mdm.query/src/com/amalto/core/storage/record/CommonStorage.java]: Why not moving all constants from Storage here and name interface StorageConstants ?

Comment (YohannTalend) [reply org.talend.mdm.query/src/com/amalto/core/storage/record/CommonStorage.java]: The dependencies in the dependencies of Storage are many to be extract.

Comment (YohannTalend) [reply org.talend.mdm.query/src/com/amalto/core/storage/record/CommonStorage.java]: Missing copyright

File: org.talend.mdm.core/pom.xml [hunk 2]
```diff
@@ -326,6 +326,11 @@
             <groupId>org.talend.mdm</groupId>
             <artifactId>org.talend.mdm.commons.core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.talend.mdm</groupId>
+            <artifactId>org.talend.mdm.query</artifactId>
+            <version>7.2.1-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (YohannTalend) [reply org.talend.mdm.core/pom.xml]: done

**Pull request merged**